app/src/main/java/org/groundplatform/android/ui/util/RotateUsingExif.kt

```diff
@@ -17,60 +17,51 @@ package org.groundplatform.android.ui.util
+ * This function ensures that images captured from camera or external sources are displayed in the
+ * correct orientation without relying on third-party libraries.
  */
-class RotateUsingExif(private val uri: Uri, private val context: Context) : Transformation {
-  override fun transform(source: Bitmap): Bitmap {
-    val orientation = getOrientationFromExif(uri)
-    val rotateDegrees = getRotationDegrees(orientation)
-    return rotateBitmap(source, rotateDegrees)
-  }
+fun loadBitmapWithCorrectOrientation(context: Context, uri: Uri): Bitmap {
+  val rotationDegrees = getRotationDegreesFromUri(context, uri)
 
-  override fun key(): String = "${uri.hashCode()}"
+  context.contentResolver.openInputStream(uri)?.use { inputStream ->
+    val bitmap = BitmapFactory.decodeStream(inputStream)
+    if (rotationDegrees == 0f) return bitmap
 
-  private fun rotateBitmap(bitmap: Bitmap, rotateDegrees: Float): Bitmap {
-    val matrix = Matrix()
-    // Rotate if rotation is non-zero.
-    if (rotateDegrees != 0f) {
-      matrix.postRotate(rotateDegrees)
-    }
-    return Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(), bitmap.getHeight(), matrix, true)
+    val matrix = Matrix().apply { postRotate(rotationDegrees) }
+    return Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
   }
 
-  /**
-   * Returns the number of degrees a photo should be rotated based on the value of its orientation
-   * EXIF tag.
-   */
-  private fun getRotationDegrees(orientation: Int): Float =
-    when (orientation) {
-      ExifInterface.ORIENTATION_UNDEFINED,
-      ExifInterface.ORIENTATION_NORMAL -> 0f
+  throw IOException("Unable to open input stream for URI: $uri")
+}
+
+/**
+ * Reads the EXIF orientation header of an image located at the given [uri] and returns the
+ * corresponding rotation in degrees.
+ *
+ * This method uses [ExifInterface] to extract metadata without fully decoding the image, which
+ * makes it memory-efficient.
+ */
+fun getRotationDegreesFromUri(context: Context, uri: Uri): Float {
+  context.contentResolver.openInputStream(uri)?.use { inputStream ->
+    val exif = ExifInterface(inputStream)
+    return when (
+      exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
+    ) {
       ExifInterface.ORIENTATION_ROTATE_90 -> 90f
       ExifInterface.ORIENTATION_ROTATE_180 -> 180f
       ExifInterface.ORIENTATION_ROTATE_270 -> 270f
-      else -> throw UnsupportedOperationException("Unsupported photo orientation $orientation")
+      else -> 0f
     }
-
-  /** Returns the EXIF orientation attribute of the JPEG image at the specified URI. */
-  private fun getOrientationFromExif(uri: Uri): Int {
-    val inputStream =
-      context.contentResolver.openInputStream(uri)
-        ?: throw IOException("Content resolver returned null for $uri")
-    val exif = ExifInterface(inputStream)
-    return exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
   }
```

Comment:
Can we preserve the existing methods `getRotationDegrees` and `getOrientationFromExif`? These lines haven't changed and would help reduce the diff count in the PR